### PR TITLE
Enable eslint jest plugin.

### PR DESCRIPTION
**Pull request by samwho**
## Description

I left a `.only` in a test in https://github.com/Budibase/budibase/pull/13266 and felt embarrassed about it, so thought I'd write a lint to help me not do that again.

Then @adrinr pointed out that the official eslint jest plugin has a check for this already, and we should use that. So I did, and I also turned on their recommended lint set.

Pro PR:
- https://github.com/Budibase/budibase-pro/pull/280

Account portal PR:
- https://github.com/Budibase/account-portal/pull/552

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -43,9 +43,24 @@
         "no-useless-escape": "off",
         "no-undef": "off",
         "no-prototype-builtins": "off",
-        "local-rules/no-budibase-imports": "error",
+        "local-rules/no-budibase-imports": "error"
+      }
+    },
+    {
+      "files": ["**/*.spec.ts"],
```
Comment (adrinr) [inline .eslintrc.json:50]: Should we use an existing plugin such as `eslint-plugin-jest` instead?

https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-focused-tests.md

Comment (adrinr) [reply .eslintrc.json:50]: We could probably also benefit form the other recommended rules

Comment (samwho) [reply .eslintrc.json:50]: I'm in the process of enabling the TypeScript recommended rules in another PR. I'll add the Jest ones in there, too. Good call!

File: .eslintrc.json [hunk 2]
```diff
@@ -43,9 +44,34 @@
...
```
Comment (adrinr) [inline .eslintrc.json:51]: Can we add the same plugins (except TS) to `spec.js` as well?

File: packages/backend-core/scripts/test.sh [hunk 3]
```diff
@@ -4,10 +4,10 @@ set -e
...
   # --runInBand performs better in ci where resources are limited
-  echo "jest --coverage --runInBand --forceExit"
-  jest --coverage --runInBand --forceExit
+  echo "jest --coverage --runInBand --forceExit $@"
+  jest --coverage --runInBand --forceExit $@
 else
   # --maxWorkers performs better in development
-  echo "jest --coverage --detectOpenHandles"
-  jest --coverage --detectOpenHandles
+  echo "jest --coverage --forceExit --detectOpenHandles $@"
```
Comment (adrinr) [inline packages/backend-core/scripts/test.sh:11]: Why is this needed? Don't we actually want to detect the handles?

File: packages/server/src/api/routes/tests/appImport.spec.ts [hunk 4]
```diff
@@ -25,8 +25,8 @@ describe("/applications/:appId/import", () => {
       .expect(200)
     const appPackage = await config.api.application.get(appId!)
     expect(appPackage.navigation?.links?.length).toBe(2)
-    expect(expect(appPackage.navigation?.links?.[0].url).toBe("/blank"))
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/appImport.spec.ts:28]: 🤯

File: packages/server/src/integrations/tests/sql.spec.ts [hunk 5]
```diff
@@ -342,25 +342,6 @@ describe("SQL query builder", () => {
     })
   })
 
-  it("should use greater than when only low range specified", () => {
```
Comment (adrinr) [inline packages/server/src/integrations/tests/sql.spec.ts:345]: Was this duplicated?

Comment (samwho) [reply packages/server/src/integrations/tests/sql.spec.ts:345]: Yes, there is a lint that fires on identical strings in `it` and `describe` blocks and there is an identical test elsewhere in this file.

File: .eslintrc.json (hunk 2 above)
Comment (samwho) [reply .eslintrc.json:51]: Good idea. Do you mind if I do that in a follow-up PR, to keep the size down on this one?

File: .eslintrc.json [hunk 6]
```diff
@@ -34,6 +34,7 @@
     },
     {
       "files": ["**/*.ts"],
+      "excludedFiles": ["qa-core/**"],
```
[the remark this answers is not in the document]
Comment (samwho) [reply .eslintrc.json:37]: Soon. Soon.

File: packages/backend-core/scripts/test.sh (hunk 3 above)
Comment (adrinr) [reply packages/backend-core/scripts/test.sh:11]: Sorry, I meant the `--forceExit`

File: .eslintrc.json (hunk 2 above)
Comment (adrinr) [reply .eslintrc.json:51]: Yes please, otherwise you will start a new rabbit hole ahah

File: packages/backend-core/scripts/test.sh (hunk 3 above)
Comment (samwho) [reply packages/backend-core/scripts/test.sh:11]: Ohh sorry, I ran these tests once and they got stuck exiting because of an open Redis client. I did this out of laziness and because it's done elsewhere.

**Review by adrinr — APPROVED** [review]
image
